### ISSUE TITLE
Why do I have to call $nextTick?

### ISSUE THREAD

**Issue by strflw** [eddyerburgh/vue-hackernews#9]
I'm just working through your book and reached chapter 4 now.
When writing the tests for the progess bar i ran into a weird problem.

When I ran the code in the browser, everything was fine. 
When start was called, the hidden data prop was set to false and the class was removed accordingly.
But, when running the tests with jest, the class didn't get removed and of obviously the test kept staying red.

Figuring around in the debugger a bit, i could see that the hidden prop was updates as expect, only the class binding didn't seem to trigger.
Considering that and the fact that it did work in the browser, i concluded it had to do how Vue works and that in the test something doesn't happen whats happening in the browser.
Im new to Vue, but I got some experience with ember, and discovered vm.$newTick.

So after making the tests async and adding a ```await wrapper.vm.$nextTick()``` after [this](https://github.com/eddyerburgh/vue-hackernews/blob/chapter-7/src/components/__tests__/ProgressBar.spec.js#L17) line
I finally got the test pass.

Since you don't mention it anywhere here and there is also no relating issue or anything, i assume this isn't the way it should work.
So I was wondering if some could explain to me whats happening.

I have to say, I had to bump the Version of firebase to ^7.15.0 because of some version incompatibilities with my system, but my Vue dep is still on ^2.5.16.
So, I don't know if that might be the culprit.

**Comment by strflw** [comment]
Ah, got it. Seems to be mandatory https://vue-test-utils.vuejs.org/guides/#using-nexttick-and-awaiting-actions
But maybe, you should mention that in the book @eddyerburgh

**Comment by eddyerburgh** [comment]
Hi @strflw, when I wrote the book Vue Test Utils performed DOM updates synchronously. Unfortunately the library was still in beta and I made the mistake of publishing the book without specifying the VTU version. I've added an errata specifying the supported version of Vue Test Utils and if there is a future edition of the book then it will include information about nextTick.

I'm sorry you had to search for this information outside of the book, I imagine it was very frustrating.
